examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java

```diff
@@ -16,22 +16,29 @@
     public static final SimpleDateFormat ADMDateFormat = new SimpleDateFormat("yyyy-MM-dd");
     public static final SimpleDateFormat ADMTimeFormat = new SimpleDateFormat("HH:mm:ss.SSS");
 
+    public static StringBuilder sbConstructor=new StringBuilder();
```

Comment:
This static StringBuilder is stateful, thus it is error-prone if used in multiple method.

Reply:
@JavierJia  Hi, the background here is our AsterixDB is not stable, and re-ingestion process has been repeated for several times, each time consuming 3-4 days. Therefore, @chenlica asks @liming-thu to look into how to accelerate this process.

For your concern, do you mean, if he uses this static stringbuilder, it might mess up the data with incorrect output? If so, I think we need to think about it more carefully and might need some more semantic tests. Thanks!

@liming-thu  Could you provide some abstract description of what you have found about this issue, e.g. which part of the code occupies how much percent of the total time of the geoTag process? So that people can understand the motivation here. And besides, a PR needs some description any way. Please refer to previous PRs, about how to write a description. Thanks!

Reply:
It will mess up for sure. One simple counter argument is that how it work in the multi threading case? 

If the context is to speed up the geotagging process, one direction I can think of is to run the command in parallel, either multi-threading or multi-process. It would improve the performance x-times faster depending the degree of parallelism. 

Yes, I agree that we can also optimize the code, which may achieve x% performance gain. If we deciding to going to this direction, I believe that we've already used some tool (e.g., [Yourkit](https://www.yourkit.com/)) to analyze the bottleneck. Could anyone share the number of the analysis?

Even it indeed the case that the builder can save a lot of time, we should pass the `builder` into the function as a parameter instead of creating a global static variable.